### PR TITLE
Small improvements

### PR DESCRIPTION
**Pull request by wendigo**
## Release notes

( ) This is not user-visible or is docs only, and no release notes are required.
( ) Release notes are required. Please propose a release note for me.
( ) Release notes are required, with the following suggested text:

```markdown
* 
```

### CODE REVIEW

File: gateway-ha/src/main/java/io/trino/gateway/ha/clustermonitor/ActiveClusterMonitor.java [hunk 1]
```diff
@@ -63,42 +68,31 @@ public ActiveClusterMonitor(
...
-                            List<ClusterStats> stats = new ArrayList<>();
-                            for (Future<ClusterStats> clusterStatsFuture : futures) {
-                                ClusterStats clusterStats = clusterStatsFuture.get();
-                                stats.add(clusterStats);
-                            }
+        log.info("Running cluster monitor with connection task delay of %d seconds", taskDelaySeconds);
+
+        scheduledFuture = scheduledExecutor.scheduleAtFixedRate(() -> {
+            try {
+                log.info("Getting the stats for the active clusters");
```
Comment (mosabua) [inline gateway-ha/src/main/java/io/trino/gateway/ha/clustermonitor/ActiveClusterMonitor.java:75]:
```suggestion
                log.info("Getting stats for all active clusters");
```

**Review by mosabua — APPROVED** [review]
Looks good. Just some minor comments. 

My main impression from this PR is a reminder that we really ought to get rid of the usage of "backend" throughout the codebase and docs and replace it with "Trino cluster" and "cluster" as necessary in the context. We agreed on doing this in a dev sync and just have not gotten around to it.. maybe we should file an issue so we dont forget.

**Comment by mosabua** [comment]
Filed the issue now - https://github.com/trinodb/trino-gateway/issues/408

File: gateway-ha/src/main/java/io/trino/gateway/ha/clustermonitor/ClusterStatsHttpMonitor.java [hunk 2]
```diff
@@ -125,9 +125,8 @@ private OkHttpClient acquireClientWithCookie(String loginUrl)
             return client;
         }
         catch (IOException e) {
-            e.printStackTrace();
+            throw new UncheckedIOException(e);
         }
-        return null;
```
Comment (oneonestar) [inline gateway-ha/src/main/java/io/trino/gateway/ha/clustermonitor/ClusterStatsHttpMonitor.java:130]: This change from using null to throwing exceptions for error handling.
The null checks need to be updated accordingly.
https://github.com/trinodb/trino-gateway/blob/4c40750eb6a1d6710526cdca5d72b2ed139d2740/gateway-ha/src/main/java/io/trino/gateway/ha/clustermonitor/ClusterStatsHttpMonitor.java#L137-L140

File: gateway-ha/src/main/java/io/trino/gateway/ha/clustermonitor/ClusterStatsHttpMonitor.java [hunk 3]
```diff
@@ -148,23 +147,18 @@ private String queryCluster(ProxyBackendConfiguration backend, String path)
...
-            }
+                    log.debug("username: %s, targetUrl: %s, cookieStore: %s", username, targetUrl, client.cookieJar().loadForRequest(HttpUrl.parse(targetUrl)));
+                    yield null;
+                }
+                default -> null;
+            };
         }
         catch (IOException e) {
-            e.printStackTrace();
+            throw new UncheckedIOException(e);
```
Comment (oneonestar) [inline gateway-ha/src/main/java/io/trino/gateway/ha/clustermonitor/ClusterStatsHttpMonitor.java:161]: I'm a bit confused on the error handling.
Why mix null and throw exception for error handling?

**Comment by mosabua** [comment]
@wendigo .. gentle ping ;-)

**Comment by wendigo** [comment]
@mosabua ? :)

**Comment by wendigo** [comment]
@oneonestar I'll go back to this PR later this week :)

**Comment by ebyhr** [comment]
@wendigo Could you update this PR?

**Comment by wendigo** [comment]
@ebyhr sure, let me find some time for it this week

**Pull request closed without merging**